src/use/uWebSockets.ts

```diff
@@ -10,12 +11,16 @@ export interface Extra {
   /**
    * The actual socket connection between the server and the client.
    */
-  readonly socket: uWS.WebSocket;
+  readonly socket: uWS.WebSocket & { upgradeReq: Request };
```

Comment:
Why include the upgrade request here too? The `request` below _is_ the upgrade request.
```suggestion
  readonly socket: uWS.WebSocket;
```

Reply:
It was there before my changes, I just wanted to save consistency between actual JS objects and TypeScript defenitions.

Reply:
No it was not. You added the `upgradeReq` haha, check the diff. 

Permalink from master:
https://github.com/enisdenjo/graphql-ws/blob/2933997cfbbe9965a52acadb0a727e6dfb87b265/src/use/uWebSockets.ts#L13

Reply:
> No it was not. You added the `upgradeReq` haha, check the diff.

I mean it was as JS object, but not as TypeScript definition.

https://github.com/enisdenjo/graphql-ws/blob/master/src/use/uWebSockets.ts#L75

Reply:
I see, but the TS at uWS claim that this:
https://github.com/enisdenjo/graphql-ws/blob/2933997cfbbe9965a52acadb0a727e6dfb87b265/src/use/uWebSockets.ts#L84
is only `uWS.WebSocket`.

[You also had to assert the typing `const socket = args[0] as uWS.WebSocket & { upgradeReq: Request }`](https://github.com/maxpain/graphql-ws/blob/f97cfb4f2b33f63efcb9a72c0ee1c3b7fdac705b/src/use/uWebSockets.ts#L96)

Reply:
Yes, for some reason developer of `uWS` doesn't give us ability to extend `uWS.WebSocket` with extra fields, passed to upgrade function :(

Reply:
Yes, for some reason developer of `uWS` doesn't give us the ability to extend `uWS.WebSocket` with extra fields, passed to upgrade function :(

So I extended it by myself here:

https://github.com/maxpain/graphql-ws/blob/f97cfb4f2b33f63efcb9a72c0ee1c3b7fdac705b/src/use/uWebSockets.ts#L96